src/System.Diagnostics.Contracts/tests/ForAllTests.cs

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Linq;
+using Xunit;
+
+namespace System.Diagnostics.Contracts.Tests
+{
+    public class ForAllTests
+    {
+        [Fact]
+        public static void ArgumentExceptions()
+        {
+            Assert.Throws<ArgumentNullException>(() => Contract.ForAll(0, 1, null));
+            Assert.Throws<ArgumentNullException>(() => Contract.ForAll<int>(null, i => true));
+            Assert.Throws<ArgumentNullException>(() => Contract.ForAll<int>(Enumerable.Empty<int>(), null));
+            Assert.Throws<ArgumentException>(() => Contract.ForAll(1, 0, i => true));
+        }
+
+        [Fact]
+        public static void EmptyInputReturnsTrue()
+        {
+            Assert.True(Contract.ForAll(Enumerable.Empty<int>(), i => true));
+            Assert.True(Contract.ForAll(-2, -2, i => true));
+            Assert.True(Contract.ForAll(1, 1, i => true));
```

Comment:
i => throw new InvalidOperationException("Didn't expect to call predicate") (for all of these)?